transformer_engine/jax/fp8.py

```diff
@@ -199,24 +200,32 @@ def update_collections(new: Collection, original: Collection) -> Collection:
         """
         Update the collections
         """
-        if not isinstance(original, FrozenDict):
-            original = FrozenDict(original)
+        assert isinstance(original, get_args(Collection))
+        assert isinstance(new, get_args(Collection))
```

Comment:
I didn't know about get_args.
Are you sure this syntax only support FrozenDict and Dict and not any other classes?
This syntax would make sure this doesn't cover any other classes and I think is easier to read:

```suggestion
        assert isinstance(original, (dict, FrozenDict))
        assert isinstance(new, (dict, FrozenDict))
```

Reply:
`(typing.Dict, <class 'flax.core.frozen_dict.FrozenDict'>)` is what get_args returned, which is equivalent to `(dict, FrozenDict)`.
The tought here is to reuse `Collection` (the datatype we defined in [fp8.py:21](https://github.com/NVIDIA/TransformerEngine/blob/2240c95bae734e9572b2023538938c9355240fac/transformer_engine/jax/fp8.py#L21)). And `get_args` is a standard approach of python to get type hints. Refer to [typing.html#typing.get_args](https://docs.python.org/3/library/typing.html#typing.get_args) for details.

Reply:
Maybe it is standard, but it is harder to understand. So fewer people will understand it at first read. This isn't great.
I don't block this PR for this. But readability is something useful.

I'll @timmoon10 finish the review/merge it.

Reply:
I agree this might make a bit worse readability, but helpful on scalability, in case that we would support more datatypes as `Collection`, such as `Collection = Union[Dict, FrozenDict, MyDefinedDict]`. Both ways we discussed are good to me. Please let me know the final decision, then I can make changes.

Reply:
Let's do `(dict, FrozenDict)` for now. If we ever make `Collection` more complicated, then we should switch to `get_args(Collection)` at that point.

Reply:
Done